packages/devtools_app/lib/src/shared/dialogs.dart

```diff
@@ -21,6 +29,94 @@ List<Widget> dialogSubHeader(ThemeData theme, String titleText) {
   ];
 }
 
+/// A standardized dialog with help text and buttons `Reset to default`,
+/// `APPLY` and `CANCEL`.
+class DialogWithDefaults extends StatelessWidget {
+  const DialogWithDefaults({
+    super.key,
+    required this.title,
+    required this.child,
+    this.helpText,
+    this.onResetDefaults,
+    this.onApply,
+    this.onCancel,
+    this.dialogWidth,
+  });
+
+  final String title;
+  final String? helpText;
+  final VoidCallback? onResetDefaults;
+  final VoidCallback? onApply;
+  final VoidCallback? onCancel;
+  final Widget child;
+  final double? dialogWidth;
+
+  @override
+  Widget build(BuildContext context) {
+    return DevToolsDialog(
+      title: _DialogWithDefaultsTitle(onResetDefaults: onResetDefaults),
+      content: Container(
+        padding: const EdgeInsets.symmetric(
+          horizontal: defaultSpacing,
+        ),
+        width: dialogWidth ?? defaultDialogWidth,
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            child,
+            if (helpText != null) ...[
+              const SizedBox(height: defaultSpacing),
+              DialogHelpText(helpText: helpText!),
+              const SizedBox(height: defaultSpacing),
+            ],
+          ],
+        ),
+      ),
+      actions: [
+        DialogApplyButton(onPressed: onApply ?? () {}),
+        DialogCancelButton(cancelAction: onCancel),
+      ],
+    );
+  }
+}
+
+class _DialogWithDefaultsTitle extends StatelessWidget {
```

Comment:
rename to _FilterDialogTitle, since again, the reset action only applies to filter dialogs